### PR TITLE
Fixes issue #146.

### PR DESCRIPTION
**Pull request by dawnmist**
* Document the alternative module.exports options.
* Discuss configuration options for Jest/Babel and when you may want to use Jest in package.json verses the 'jest' function in config-overrides.js
* Document how to customise the configuration for the Webpack Dev Server.

### CODE REVIEW

**Comment by Guria** [comment]
Awesome. Still needs some review to fix minor grammar issues. But content very descriptive indeed.

I would also add another way to manage overrides in folder instead of single file by creating `config-overrides/index.js`. See [my boilerplate](https://github.com/Guria/rewired-ts-boilerplate/tree/master/config-overrides) as reference.

Probably we may note that currently it is very hard to override entry-point. In CRA it [defined in path.js](https://github.com/facebookincubator/create-react-app/blob/2e82ebb3371731a5c4e346f310848ddb23fd0976/packages/react-scripts/config/paths.js#L55).
But we can't just patch it, because [require.cache is dropped in env.js](https://github.com/facebookincubator/create-react-app/blob/2e82ebb3371731a5c4e346f310848ddb23fd0976/packages/react-scripts/config/env.js#L16). We can't just override entry in webpack config too, because entry defined in paths is [used by other scripts](https://github.com/facebookincubator/create-react-app/blob/2e82ebb3371731a5c4e346f310848ddb23fd0976/packages/react-scripts/scripts/build.js#L47).
In typescript boilerplate I [had to leave index.js as entry point and just [import index.ts](https://github.com/Guria/rewired-ts-boilerplate/blob/b9e8f95/src/index.js) from it as a workaround. Another way I found just now is to short-circuit [checkRequiredFiles](https://github.com/facebookincubator/create-react-app/blob/2e82ebb3371731a5c4e346f310848ddb23fd0976/packages/react-dev-utils/checkRequiredFiles.js) utility to return always true.

**Comment by dawnmist** [comment]
Good point on overriding the entry point - I worked around it by using react-scripts-ts instead of react-scripts (which meant it was set up for index.ts by default), but that means that I'm using the --scripts-version option for react-app-rewired which is also not documented in the Readme file. I'll add in some documentation for using a config-overrides directory, description on why its hard to override the entry-point with , and on using --scripts-version to set react-app-rewired to use a custom version of react-scripts.

**Comment by Guria** [comment]
In --scripts-version docs we can mention file paths that we rely on in our script:
* config/env
* config/webpack.config.dev
* config/webpack.config.prod
* config/webpackDevServer.config.js
* scripts/build
* scripts/start
* scripts/test
* scripts/utils/createJestConfig

File: README.md [hunk 1]
```diff
@@ -183,23 +183,121 @@ module.exports = {
...
+    "start": "react-app-rewired start --scripts-version react-scripts-ts",
+    "build": "react-app-rewired build --scripts-version react-scripts-ts",
+    "test": "react-app-rewired test --scripts-version react-scripts-ts --env=jsdom",
+    "eject": "react-scripts eject"
+  }
+}
+```
+
+React-app-rewired requires a custom react-scripts package to provide the following files:
+* config/env
```
Comment (Guria) [inline README.md]: js extension were lost

File: README.md [hunk 2]
```diff
@@ -155,6 +155,149 @@ function createRewire(requiredParams){
 module.exports = createRewire;
 ```
 
+## Expanded Configuration Options
+By default, the `override-config.js` file exports a single function to use when customising the webpack configuration for compiling your react app in development or production mode. It is possible to instead export an object from this file that contains up to three fields, each of which is a function. This alternative form allows you to also customise the configuration used for Jest (in testing), and for the Webpack Dev Server itself.
+
+```javascript
+module.exports = {
```
Comment (Guria) [inline README.md]: now we have 2 almost similar examples: here and in `4) Example object-style config-overrides.js` section

Comment (dawnmist) [reply README.md]: Oops. I was working further down in the file, and had forgotten that I had already created an initial example. I've merged the details for the two into the first of the examples.

**Pull request merged**